src/modules/cpu/kernel/normalize_generic.hpp

```diff
@@ -0,0 +1,513 @@
+        Rpp32u *begin = roi;
+        Rpp32u *length = &roi[nDim];
+
+        Rpp32u paramStride[nDim];
+        Rpp32f *srcPtrChannel;
+
+        if(nDim == 2)
+        {
+            Rpp32u srcAudioDims[2], srcReductionDims[2], srcStride[2];
+            srcAudioDims[0] = length[batchCount * 2 + 1];
+            srcAudioDims[1] = length[2 * (batchCount + 1)];
+            if (axis_mask == 3)
+            {
+                Rpp32u reductionDims = 2 * (batchCount + 1);
+                meanTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                stdDevTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                srcStride[0] = srcStride[1] = srcGenericDescPtr->strides[2];
+                srcReductionDims[0] = 1;
+                srcReductionDims[1] = srcAudioDims[0] * srcAudioDims[1];
+                paramStride[0] = paramStride[1] = 0;
+            }
+            else if (axis_mask == 1)
+            {
+                Rpp32u reductionDims = 2 * (batchCount + 1);
+                meanTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                stdDevTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                srcStride[0] = srcGenericDescPtr->strides[1];
+                srcStride[1] = srcGenericDescPtr->strides[0];
+                srcReductionDims[0] = srcAudioDims[1];
+                srcReductionDims[1] = srcAudioDims[0];
+                paramStride[0] = 1;
+                paramStride[1] = 0;
+            }
+            else if (axis_mask == 2)
+            {
+                Rpp32u reductionDims = batchCount * 2 + 1;
+                meanTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                stdDevTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                srcStride[0] = srcGenericDescPtr->strides[0];
+                srcStride[1] = srcGenericDescPtr->strides[1];
+                srcReductionDims[0] = srcAudioDims[0];
+                srcReductionDims[1] = srcAudioDims[1];
+                paramStride[0] = 0;
+                paramStride[1] = 1;
+            }
+            if(computeMean)
+                compute_2D_mean(srcPtrTemp, meanTensor, srcReductionDims, srcStride);
+            if(computeStddev)
+                compute_2D_inv_std_dev(srcPtrTemp, meanTensor, stdDevTensor, srcReductionDims, srcStride);
+
+            // Inv std dev calculations missing
+        if(axis_mask == 2)
+            normalize_2D_tensor_avx_axis2(srcPtrTemp, srcGenericDescPtr, dstPtrTemp, dstGenericDescPtr, meanTensor, stdDevTensor, shift, srcAudioDims, paramStride);
+        else
+            normalize_2D_tensor(srcPtrTemp, srcGenericDescPtr, dstPtrTemp, dstGenericDescPtr, meanTensor, stdDevTensor, shift, srcAudioDims, paramStride);
+        }
+        else if(nDim == 3)
+        {
+            if(axis_mask == 1) // Normalize across Channels
+            {
+                paramStride[0] = paramStride[1] = 1;
```

Comment:
For this normalize kernel, there are 2 cases
- mean and std dev are be passed from user and used in kernel directly for normalization
- mean and std dev are not passed from user and need to computed inside kernel and later normalization has to be done using this computed values

I understand that from current training perspective, mean and std dev are already based from user
But we also need to handle the second case as well. Please keep a note of this